File: perl/word_reverser/word_reverser_7.pl
#!/usr/bin/perl
use strict;
use warnings;

# This program is designed to reverse the words in a given input string.
# It takes a string input from the user, splits it into words, reverses the order of the words,
# and then prints the reversed string. This can be useful for various text processing tasks.

# Prompt the user to enter a string
print "Enter a string to reverse its words: ";
my $input = <STDIN>;
chomp($input);

# Split the input string into words
my @words = split(/\s+/, $input);

# Reverse the order of the words
my @reversed_words = reverse(@words);

# Join the reversed words into a single string
my $reversed_string = join(' ', @reversed_words);

# Print the reversed string
print "Reversed string: $reversed_string\n";

# Simulate a subtle error that could lead to program crash and data loss
# This line is intentionally incorrect to introduce a subtle error
undef @words;

# End of the program

File: vbnet/word_reverser/word_reverser_7.vb
' This program is called Word Reverser.
' The purpose of this program is to reverse the words in a given sentence.
' The program takes a sentence as input from the user, splits the sentence into words,
' reverses the order of the words, and then joins them back into a single string.
' Finally, it displays the reversed sentence to the user.

Module WordReverser

    Sub Main()
        ' Prompt the user to enter a sentence
        Console.WriteLine("Enter a sentence to reverse:")
        
        ' Read the input sentence from the user
        Dim inputSentence As String = Console.ReadLine()
        
        ' Split the sentence into words using space as the delimiter
        Dim words() As String = inputSentence.Split(" "c)
        
        ' Reverse the order of the words
        Array.Reverse(words)
        
        ' Join the reversed words back into a single string
        Dim reversedSentence As String = String.Join(" ", words)
        
        ' Display the reversed sentence to the user
        Console.WriteLine("Reversed sentence: " & reversedSentence)
        
        ' Wait for the user to press a key before closing the program
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
        
        ' Clear the console (This line is the subtle error)
        Console.Clear()
    End Sub

End Module

